esmvalcore/preprocessor/_derive/intppgt.py

```diff
@@ -0,0 +1,89 @@
+"""Derivation of variable `intppgt`."""
+import iris
+import numpy as np
+
+from ._baseclass import DerivedVariableBase
+
+
+def calculate_total_flux(intpp_cube, cube_area):
+    """
+    Calculate the area of unmasked cube cells.
+
+    Requires a cube with two spacial dimensions. (no depth coordinate).
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Data Cube
+    cube_area: iris.cube.Cube
+        Cell area Cube
+
+    Returns
+    -------
+    numpy.array:
+        An numpy array containing the total flux of CO2.
+    """
+    data = []
+    times = intpp_cube.coord('time')
+
+    intpp_cube.data = np.ma.array(intpp_cube.data)
+    for time_itr in np.arange(len(times.points)):
+
+        total_flux = intpp_cube[time_itr].data * cube_area.data
+
+        total_flux = np.ma.masked_where(intpp_cube[time_itr].data.mask,
+                                        total_flux)
+        data.append(total_flux.sum())
+
+    ######
+    # Create a small dummy output array
+    data = np.array(data)
+    return data
+
+
+class DerivedVariable(DerivedVariableBase):
+    """Derivation of variable `intppgt`."""
+
+    # Required variables
+    required = [
+        {
+            'short_name': 'intpp',
+            # 'mip': 'Omon',
+            'fx_files': [
+                'areacello',
+            ],
+        },
+    ]
+
+    @staticmethod
+    def calculate(cubes):
+        """Compute longwave cloud radiative effect."""
+        intpp_cube = cubes.extract_strict(
+            iris.Constraint(name='net_primary_mole_productivity_of_carbon_by_'
+                                 'phytoplankton'))
+
+        try:
+            cube_area = cubes.extract_strict(iris.Constraint(name='cell_area'))
+        except iris.exceptions.ConstraintMismatchError:
```

Comment:
If this happens, `cube_area` will be undefined and the code below will fail, so I think it is not a good idea to let this `pass`